pkg/kubelet/kubelet_node_status.go

```diff
@@ -91,7 +93,16 @@ func (kl *Kubelet) tryRegisterWithAPIServer(node *v1.Node) bool {
 		return true
 	}
 
-	if !apierrors.IsAlreadyExists(err) {
+	switch {
+	case apierrors.IsAlreadyExists(err):
+		// Node already exists, proceed to reconcile node.
```

Comment:
@SergeyKanzhelev how do we guarantee that the node that already exists is the right one?

does this mean I can have two kubelets with the same name?

what happen with spot vms that are different vms wtih the same name, do they not need to re-register and update the corresponding fields? I'm thinking about nodeIPAM that indeed changes between vms and is inmutable

Reply:
this is an old logic, there is no validation on whether it is the same node object. Nodes are guarantee to have different names. We just reuse and reconcile the node if it has the same name. Realistically it is hard to distinguish various situations where this may return that the node already exists. It may be kubelet restart, retry after timeout, reusing "old" node, etc.

This PR doesn't make situation neither better nor worth w.r.t. reusing old node objects

Reply:
doh, before was `if !apierrors.IsAlreadyExists(err)` , sorry for the noise, my bad